tests/utils/utils.go

```diff
@@ -944,3 +947,81 @@ func SendExampleCrossChainMessageAndVerify(
 		Expect(currMessage).ShouldNot(Equal(message))
 	}
 }
+
+func InitChainConfig(
+	networkID uint32,
+	subnet interfaces.SubnetTestInfo,
+	teleporterAddress common.Address,
+) (*avalancheWarp.UnsignedMessage, string) {
+	unsignedMessage := CreateOffChainRegistryMessage(networkID, subnet, teleporterregistry.ProtocolRegistryEntry{
+		Version:         big.NewInt(2),
+		ProtocolAddress: teleporterAddress,
+	})
+	offChainMessage := hexutil.Encode(unsignedMessage.Bytes())
+	log.Info("Adding off-chain message to Warp chain config",
+		"messageID", unsignedMessage.ID(),
+		"blockchainID", subnet.BlockchainID.String())
+
+	return unsignedMessage, fmt.Sprintf(`{
+    "warp-api-enabled": true, 
+    "warp-off-chain-messages": ["%s"],
+	"log-level": "debug",
+    "eth-apis":["eth","eth-filter","net","admin","web3",
+                "internal-eth","internal-blockchain","internal-transaction",
+                "internal-debug","internal-account","internal-personal",
+                "debug","debug-tracer","debug-file-tracer","debug-handler"]
+	}`, offChainMessage)
+}
+
+func CreateOffChainRegistryMessage(
+	networkID uint32,
+	subnet interfaces.SubnetTestInfo,
+	entry teleporterregistry.ProtocolRegistryEntry,
+) *avalancheWarp.UnsignedMessage {
+	sourceAddress := []byte{}
+	payloadBytes, err := teleporterregistry.PackTeleporterRegistryWarpPayload(entry, subnet.TeleporterRegistryAddress)
+	Expect(err).Should(BeNil())
+
+	addressedPayload, err := payload.NewAddressedCall(sourceAddress, payloadBytes)
+	Expect(err).Should(BeNil())
+
+	unsignedMessage, err := avalancheWarp.NewUnsignedMessage(
+		networkID,
+		subnet.BlockchainID,
+		addressedPayload.Bytes())
+	Expect(err).Should(BeNil())
+
+	return unsignedMessage
+}
+
+func DeployNewTeleporterVersion(
+	ctx context.Context,
+	network interfaces.LocalNetwork,
+	fundedKey *ecdsa.PrivateKey,
+	teleporterByteCodeFile string,
+) common.Address {
+	contractCreationGasPrice := (&big.Int{}).Add(deploymentUtils.GetDefaultContractCreationGasPrice(), big.NewInt(1))
```

Comment:
note that when this gas price for the tx increases, the funds transferred to the deployer to `DeployTeleporterContracts` will also need to increase

Reply:
The deployer address is funded in `network.DeployTeleporterContracts`

Reply:
As in right now the `DeployTeleporterContracts` hardcodes to 11avax, but if multiple `DeployNewTeleporterVersion`s are called, each time it'd require more funds, but it's still hardcoded at 11avax so would end up failing

Reply:
I don't think we'll ever realistically hit that case in our tests. The gas price in the keyless tx is set to 2500 nAVAX, and the gas limit to 4000000, which are intentionally high to prevent network congestion from causing the tx to fail.